packages/components/src/components/post-language-option/post-language-option.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Method,
+  Prop,
+  Watch,
+} from '@stencil/core';
+import { checkEmptyOrType, checkType } from '@/utils';
+import { version } from '@root/package.json';
+
+/**
+ * @slot default - Slot for placing the content inside the anchor or button.
+ */
+@Component({
+  tag: 'post-language-option',
+  styleUrl: 'post-language-option.scss',
+  shadow: true,
+})
+export class PostLanguageOption {
+  @Element() host: HTMLPostLanguageOptionElement;
+
+  /**
+   *  The ISO 639 language code, formatted according to [RFC 5646 (also known as BCP 47)](https://datatracker.ietf.org/doc/html/rfc5646). For example, "de".
+   */
+  @Prop() code!: string;
+
+  @Watch('code')
+  validateCode(value = this.code) {
+    checkType(
+      value,
+      'string',
+      'The "code" property of the post-language-option component must be a string.',
+    );
+  }
+
+  /**
+   * If set to `true`, the language option is considered the current language for the page.
+   */
+  @Prop({ mutable: true, reflect: true }) active: boolean;
+
+  @Watch('active')
+  validateActiveProp(value = this.active) {
+    checkEmptyOrType(
+      value,
+      'boolean',
+      'The "active" property of the post-language-option component must be a boolean value.',
+    );
+  }
+
+  /**
+   * The full name of the language. For example, "Deutsch".
+   */
+  @Prop() name: string;
```

Comment:
The name is a required prop, even if not active. It will be read when choosing options, thanks to the language code in the appropriate language.